test/Tests/src/Semantic/Meta_Spec.enso

```diff
@@ -102,6 +102,15 @@ spec =
             1.0.is_a Integer . should_be_false
             1.0.is_a Text . should_be_false
 
+            Number.nan.is_a Number . should_be_true
+            Number.nan.is_a Decimal . should_be_true
+            Number.nan.is_a Nan . should_be_true
+            Number.nan.is_a Integer . should_be_false
+
+            (1.0%0.0).is_a Number . should_be_true
```

Comment:
Should `NaN` be also added into `ValuesGenerator` and used in some tests?

Reply:
Yes, just working on it. `MetaObjectTest` is failing because of it anyway.